ext2/ext2.c

```diff
@@ -121,9 +121,10 @@ int ext2_lookup(ext2_t *fs, id_t id, const char *name, size_t len, oid_t *res, o
 
 			obj = ext2_obj_get(fs, res->id);
 			if (obj == NULL) {
+				mutexUnlock(dir->lock);
```

Comment:
uhh, the current code structure is inconsistient - still `do {} while (0)` is the roundabout way for error handling to have only single point of deinit/exit (similar to `try...catch`, failure paths via `goto`, etc).

I would like to avoid ^C^V deinit/exit in this kind of block if possible as it further degrades the code quality.

Maybe call instead directly the function `_ext2_dir_remove` to remove the direntry (needs to be called with `dir->lock` which we already have here :)).

Reply:
True, direct call is much better, haven't thought about it. Corrected